webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx

```diff
@@ -126,7 +126,8 @@ const ExperimentCheckpoints: React.FC<Props> = ({ experiment, pageRef }: Props)
       setSelectedCheckpoints(checkpoints);
       registerModal.open();
     },
-    [registerModal],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
im worried about doing this. its not recommended to disable the dependency array.
i think we can do this without disabling it
```diff
diff --git a/webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx b/webui/react/src/pages/ExperimentDetails/Exper>
index 23edfa766eb..a320485e21f 100644
--- a/webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx
+++ b/webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx
@@ -75,16 +75,20 @@ const ExperimentCheckpoints: React.FC<Props> = ({ experiment, pageRef }: Props)
   });
 
   const modelCreateModal = useModal(ModelCreateModal);
-  const registerModal = useModal(RegisterCheckpointModal);
+  const {
+    Component: RegisterModalComponent,
+    open: registerModalOpen,
+    close: registerModalClose,
+  } = useModal(RegisterCheckpointModal);
 
   const handleOnCloseCreateModel = useCallback(
     (modelName?: string) => {
       if (modelName) {
         setSelectedModelName(modelName);
-        registerModal.open();
+        registerModalOpen();
       }
     },
-    [setSelectedModelName, registerModal],
+    [setSelectedModelName, registerModalOpen],
   );
 
   const clearSelected = useCallback(() => {
@@ -124,9 +128,9 @@ const ExperimentCheckpoints: React.FC<Props> = ({ experiment, pageRef }: Props)
   const handleRegisterCheckpoint = useCallback(
     (checkpoints: string[]) => {
       setSelectedCheckpoints(checkpoints);
-      registerModal.open();
+      registerModalOpen();
     },
-    [registerModal],
+    [registerModalOpen],
   );
 
   const handleDelete = useCallback(async (checkpointUuids: string[]) => {
@@ -393,9 +397,9 @@ const ExperimentCheckpoints: React.FC<Props> = ({ experiment, pageRef }: Props)
         )}
       </Section>
       <modelCreateModal.Component onClose={handleOnCloseCreateModel} />
-      <registerModal.Component
+      <RegisterModalComponent
         checkpoints={selectedCheckpoints ?? []}
-        closeModal={registerModal.close}
+        closeModal={registerModalClose}
         modelName={selectedModelName}
         models={models}
         openModelModal={modelCreateModal.open}
```